syndicate/utils.py

```diff
@@ -0,0 +1,247 @@
+    # Default to an empty dictionary if no results have yet been persisted.
+    return json.loads(os.getenv('SYNDICATE_POSTS', '{}'))
+
+# Memoize authentication and repo fetching.
+@functools.lru_cache(maxsize=1)
+def repo():
+    """
+    (MEMOIZED) Returns an authenticated reference to a repository object for the
+    repository this Github action is running in.
+    @see https://pygithub.readthedocs.io/en/latest/github_objects/Repository.html#github.Repository.Repository
+    """
+    assert os.getenv("GITHUB_TOKEN"), "missing GITHUB_TOKEN"
+    assert os.getenv("GITHUB_REPOSITORY"), "missing GITHUB_REPOSITORY"
+
+    gh = Github(os.getenv("GITHUB_TOKEN"))
+    return gh.get_repo(os.getenv("GITHUB_REPOSITORY"))
+
+def parent_sha():
+    """
+    Returns the git SHA to use as parent for any commits generated by this
+    Github workflow step.
+    """
+    assert os.getenv("GITHUB_SHA"), "missing GITHUB_SHA"
+    return os.getenv('SYNDICATE_SHA', os.getenv("GITHUB_SHA"))
+
+def get_trigger_payload():
+    """
+    Returns a list of lightweight File objects describing each of the modified
+    files in the commit that triggered this Github workflow.
+    @see https://pygithub.readthedocs.io/en/latest/github_objects/File.html#github.File.File
+    """
+    assert os.getenv("GITHUB_SHA"), "missing GITHUB_SHA"
+    # NOTE
+    # Explicitly using GITHUB_SHA to ensure we always have access to the changed
+    # files even if other steps generate commits.
+    return repo().get_commit(os.getenv("GITHUB_SHA")).files
+
+def file_contents(filepath):
+    """
+    Returns a `ContentFile` object of the matching the given path in latest known
+    commit to this repo.
+    @see https://pygithub.readthedocs.io/en/latest/github_objects/ContentFile.html#github.ContentFile.ContentFile
+    @see :func:`~syndicate.utils.parent_sha`
+    """
+    # NOTE
+    # Using the latest known commit to ensure we capture any modifications made
+    # to the post frontmatter by previous actions.
+    return repo().get_contents(filepath, ref=parent_sha())
+
+def get_posts(post_dir=os.getenv('SYNDICATE_POST_DIR', 'posts')):
+    """
+    Returns the latest known :func:`~syndicate.utils.file_contents` of the files
+    added and modified in the commit that triggered this Github workflow.
+    """
+    files = get_trigger_payload()
+    assert files, "target commit was empty"
+
+    posts = [file for file in files if file.filename.startswith(post_dir)]
+    return [
+        file_contents(post.filename)
+        for post in posts
+        if post.status != 'deleted'  # ignore deleted files
+    ]
+
+def fronted(post):
+    """
+    Returns the :py:class:`frontmatter.Post` representation of the given
+    :func:`~syndicate.utils.file_contents` object.
+
+    If `post` is actually already a `frontmatter.Post`, this is a no-op.
+    """
+    assert post, "missing post"
+    if type(post) == frontmatter.Post:
```

Comment:
use `isinstance(object, other_object)` instead of `type(object) == object`

Reply:
Done in https://github.com/dabrady/syndicate/pull/1/commits/9ade4e12ad43c16807f6ab4517de4d7c90ba9c6d